repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/services/datastore/RemoteDatastoreServiceTest.groovy

```diff
@@ -0,0 +1,172 @@
+
+        reposeOneEndpoint = "http://localhost:$repose1Port"
+        reposeTwoEndpoint = "http://localhost:$repose2Port"
+        remoteDatastoreEndpoint = "http://localhost:$datastorePort"
+
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(targetPort)
+
+        def launcherAndLog1 = startRepose('repose1', repose1Port, targetPort, datastorePort, true)
+        repose1 = launcherAndLog1.ReposeValveLauncher
+
+        def launcherAndLog2 = startRepose('repose2', repose2Port, targetPort, datastorePort, true)
+        repose2 = launcherAndLog2.ReposeValveLauncher
+
+        def launcherAndLogR = startRepose('remote', remotePort, targetPort, datastorePort, false)
+        remote = launcherAndLogR.ReposeValveLauncher
+
+        waitUntilReadyToServiceRequests(launcherAndLog1.ReposeLogSearch)
+        waitUntilReadyToServiceRequests(launcherAndLog2.ReposeLogSearch)
+        waitUntilReadyToServiceRequests(launcherAndLogR.ReposeLogSearch)
+    }
+
+    def cleanupSpec() {
+        deproxy?.shutdown()
+        repose1?.stop()
+        repose2?.stop()
+        remote?.stop()
+    }
+
+    static def startRepose(String subName, int reposePort, int targetPort, int datastorePort, boolean client) {
+        def testProperties = new TestProperties(RemoteDatastoreServiceTest.canonicalName.replace('.', '/') + "/$subName")
+        testProperties.setReposePort(reposePort)
+        testProperties.setTargetPort(targetPort)
+        def reposeConfigProvider = new ReposeConfigurationProvider(testProperties)
+        def reposeValveLauncher = new ReposeValveLauncher(reposeConfigProvider, testProperties)
+        reposeValveLauncher.enableDebug()
+        def reposeLogSearch = new ReposeLogSearch(testProperties.getLogFile())
+        reposeLogSearch.cleanLog()
+        def params = testProperties.getDefaultTemplateParams()
+        def runDir = testProperties.runDirectory
+        params.put('repose.artifact.directory', "$runDir/artifacts")
+        params.put('datastorePort', datastorePort)
+        reposeValveLauncher.configurationProvider.applyConfigs("common", params)
+        reposeValveLauncher.configurationProvider.applyConfigs("features/services/datastore/remote", params)
+        def type = client ? "client" : "datastore"
+        reposeValveLauncher.configurationProvider.applyConfigs("features/services/datastore/remote/$type", params)
+        reposeValveLauncher.start(false, false, "repose", type)
+        def artifacts = (new File(runDir, "artifacts")).listFiles(
+                ((FilenameFilter) new WildcardFileFilter("filter-bundle-*.ear")))
+        artifacts.each { file ->
+            def fileOrig = new File(testProperties.reposeHome, "artifacts/${file.name}")
+            def fileNew = new File(runDir, "$subName/artifacts/${file.name}")
+            fileNew.parentFile.mkdirs()
+            def fileDest = new FileOutputStream(fileNew)
+            Files.copy(fileOrig.toPath(), fileDest)
+        }
```

Comment:
Oh, and yeah, not exploding the artifacts would be great (because really, it's unnecessary).

However, it appears that you are, in fact, using the `../repose_home/artifacts/` directory as I expected. The deployment directory is even set to `../respose_home` rather than `../test-homes/...`. I don't see why that wouldn't work, though. That is, I don't see what requires us to provide each Repose instance its own copy of the artifacts. All Repose is going to do is find the artifact files in the artifact directory, extract them to a deployment directory (named with a UUID generated by Repose, and so very likely unique between instances of Repose), and use the URL of that deployment directory to load classes.

```
~/dev/repose  REP-5234_RemoteDatastore ✔                                                                                                                             2d  ⍉
▶ cat repose-aggregator/tests/functional-tests/build/test-homes/features/services/datastore/RemoteDatastoreServiceTest/repose1/configs/container.cfg.xml
<?xml version="1.0" encoding="UTF-8"?>
<repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
    <deployment-config>
        <deployment-directory auto-clean="true">
            /Users/dami6356/dev/repose/repose-aggregator/tests/functional-tests/build/repose_home
        </deployment-directory>
        <artifact-directory check-interval="1000">
            /Users/dami6356/dev/repose/repose-aggregator/tests/functional-tests/build/repose_home/artifacts
        </artifact-directory>
        <logging-configuration href="file:///Users/dami6356/dev/repose/repose-aggregator/tests/functional-tests/build/test-homes/features/services/datastore/RemoteDatastoreServiceTest/repose1/configs/log4j2-test.xml"/>
    </deployment-config>
</repose-container>
```

Since the test works, I'm thinking we don't need to do this copy at all. If we are going to do it, though, we need to fix it.